bash/sanity_checks.bash

```diff
@@ -48,13 +54,58 @@ function Perform_Internal_Sanity_Checks()
     Internally_Ensure_Given_Files_Exist \
         'These Python scripts should be shipped within the hybrid handler codebase.' '--' \
         "${HYBRID_external_python_scripts[@]}"
-    Internally_Ensure_Given_Files_Exist \
-        'These base configuration files should be shipped within the hybrid handler codebase.' '--' \
-        "${HYBRID_software_base_config_file[@]}"
+    for key in "${!HYBRID_software_base_config_file[@]}"; do
+        if [[ "$key" != "Sampler" ]]; then
+            Internally_Ensure_Given_Files_Exist \
+                'These base configuration files should be shipped within the hybrid handler codebase.' '--' \
+                "${HYBRID_software_base_config_file[$key]}"
+        fi
+    done
 }
 
 #===================================================================================================
 
+function __static__Ensure_Valid_Module_Given()
+{
+    if [[ ! "${HYBRID_module[Sampler]}" =~ ^(SMASH|FIST)$ ]]; then
+        exit_code=${HYBRID_fatal_logic_error} Print_Fatal_And_Exit \
+            'The module specified for the Sampler run is not valid.' \
+            'Valid modules are: ' --emph 'SMASH' ' and ' --emph 'FIST' '.'
+    fi
+    return 0
+}
+
+function __static__Ensure_Additional_Paths_Given_For_Sampler()
+{
+    if [[ "${HYBRID_module[Sampler]}" = 'FIST' ]]; then
+        Ensure_Given_Files_Exist \
+            'Some needed file for Thermal-FIST sampler was not specified in the configuration file' \
+            -- "${HYBRID_fist_module[Particle_file]}" "${HYBRID_fist_module[Decays_file]}"
+    fi
+}
+
+function __static__Set_Sampler_Input_Key_Paths()
+{
+    declare -rgA HYBRID_sampler_input_key_default_paths=(
```

Comment:
Since reading this I asked myself "Why can't this array be declared in the global variables file?", I would add here an amplification comment to explain this. If I am correct, it is because the `particle_list_file` and the `decays_list_file` could be changed by the user in the handler config file through the `HYBRID_fist_module` array. Hence, for convenience, we put here everything. Is it so or am I missing something else? 🤔

Reply:
This is indeed the correct reason, I will point this out accordingly.